RawSpeed/NikonDecompressor.cpp

```diff
@@ -70,6 +67,10 @@ void NikonDecompressor::DecompressNikon(ByteStream *metadata, uint32 w, uint32 h
   pUp2[0] = metadata->getShort();
   pUp2[1] = metadata->getShort();
 
+  vector<ushort16> curve(65536);
+  for (size_t i = 0; i < curve.size(); i++)
```

Comment:
Absolutely. I was simply making sure the buffer is as large as it was before. Only that I now realise, it was never meaningful to have it that large, because the second half of the buffer was never used. I'll add a patch reducing the size to 0x8000.